Update eden_pipeline.py

yanivbi opened the pull request:
This pull request includes:
(1) Performance enhancement: better worst-case compression ratio, by slicing an input vector to RHT-compatible power of two length slices. (2) Shared randomness that is HW and SW Independent: Adding a stand-alone shared PRNG code to achieve consistent shared randomness over different HW platforms and SW versions.

Co-Authored-By: Shay Vargaftik <16802840+shayvar@users.noreply.github.com>

yanivbi committed:
Update eden_pipeline.py

This commit includes:
(1) Performance enhancement: better worst-case compression ratio, by slicing an input vector to RHT-compatible power of two length slices.
(2) Shared randomness that is HW and SW Independent: Adding a stand-alone shared PRNG code to achieve consistent shared randomness over different HW platforms and SW versions.

Co-Authored-By: Shay Vargaftik <16802840+shayvar@users.noreply.github.com>
Signed-off-by: Yaniv Ben-Itzhak <yaniv.benizhak@gmail.com>

psfoley commented:
Thanks for the contribution @yanivbi and @shayvar! Once you fix the [flake8 formatting issues](https://github.com/intel/openfl/actions/runs/3900633259/jobs/6661519371), and [retroactively sign your commits](https://github.com/intel/openfl/pull/681/checks?check_run_id=10599617292) (`git commit -s...`), this looks good to merge

yanivbi committed:
Update eden_pipeline.py

This commit includes:
(1) Performance enhancement: better worst-case compression ratio, by slicing an input vector to RHT-compatible power of two length slices.
(2) Shared randomness that is HW and SW Independent: Adding a stand-alone shared PRNG code to achieve consistent shared randomness over different HW platforms and SW versions.

Signed-off-by: Yaniv Ben-Itzhak <yaniv.benizhak@gmail.com>
Co-Authored-By: Shay Vargaftik <16802840+shayvar@users.noreply.github.com>
Signed-off-by: Yaniv Ben-Itzhak <yaniv.benizhak@gmail.com>
